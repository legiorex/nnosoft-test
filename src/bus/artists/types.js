export const types = {
    // Sync
    FILL_ARTISTS: 'FILL_ARTISTS',

    // Async
    FETCH_ARTISTS_ASYNC: 'FETCH_ARTISTS_ASYNC',
};
