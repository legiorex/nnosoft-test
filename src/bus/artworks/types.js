export const types = {
    // Sync
    FILL_ARTWORKS: 'FILL_ARTWORKS',

    // Async
    FETCH_ARTWORKS_ASYNC: 'FETCH_ARTWORKS_ASYNC',
};
