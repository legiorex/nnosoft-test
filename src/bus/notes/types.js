export const types = {
    // Sync
    ADD_NOTE:          'ADD_NOTE',
    WRITE_NOTES_LOCAL: 'WRITE_NOTES_LOCAL',
    SEND_NOTES:        'SEND_NOTES',

};
