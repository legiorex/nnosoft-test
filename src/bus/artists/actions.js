// Types
import { types } from "./types";

// Instruments
export const artistsActions = {

    // Sync

    fillPosts: (artists) => {
        return {
            type:    types.FILL_POSTS,
            payload: artists,
        };
    },

    // Async

    fetchArtists: () => {
        return {
            type: types.FETCH_ARTISTS_ASYNC,
        };
    },
};
