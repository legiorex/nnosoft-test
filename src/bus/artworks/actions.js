// Types
import { types } from "./types";

export const artworksActions = {

    // Sync

    fillArtworks: (artworks) => {
        return {
            type:    types.FILL_ARTWORKS,
            payload: artworks,
        };
    },

    // Async

    fetchArtworks: () => {
        return {
            type: types.FETCH_ARTWORKS_ASYNC,
        };
    },
};
