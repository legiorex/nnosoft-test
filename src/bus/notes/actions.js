// Types
import { types } from "./types";

export const notesActions = {

    // Sync

    addNote: (note) => {
        return {
            type:    types.ADD_NOTE,
            payload: note,
        };
    },

    writeNotesLocal: () => {
        return {
            type: types.WRITE_NOTES_LOCAL,
        };
    },
    sendNotes: (notes) => {
        return {
            type:    types.SEND_NOTES,
            payload: notes,
        };
    },

};
