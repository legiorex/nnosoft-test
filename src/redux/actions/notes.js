export const ADD_NOTE = "ADD_NOTE";

export function addNote (text, afterIndex) {
    return {
        type: ADD_NOTE,
        text,
        afterIndex,
    };
}
