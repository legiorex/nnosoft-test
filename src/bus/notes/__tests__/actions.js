// Types
import { types } from '../types';

// Actions
import { notesActions } from '../actions';

describe(' notes actions', () => {
    test('addNote', () => {
        expect(notesActions.addNote(__.note)).toMatchSnapshot();
    });

    test('writeNotesLocal', () => {
        expect(notesActions.writeNotesLocal()).toMatchSnapshot();
    });
    test('sendNotes', () => {
        expect(notesActions.sendNotes(__.notes)).toMatchSnapshot();
    });
});
