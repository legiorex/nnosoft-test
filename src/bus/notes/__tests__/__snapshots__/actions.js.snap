// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` notes actions addNote 1`] = `
Object {
  "payload": Object {
    "indexNote": "0",
    "text": "Hello!",
  },
  "type": "ADD_NOTE",
}
`;

exports[` notes actions sendNotes 1`] = `
Object {
  "payload": Array [
    "test",
  ],
  "type": "SEND_NOTES",
}
`;

exports[` notes actions writeNotesLocal 1`] = `
Object {
  "type": "WRITE_NOTES_LOCAL",
}
`;
